### PR TITLE
Adds a blacklist for putting guns in turrets

### LINKED ISSUE #75197: Duelling pistols can be used in a turret to bypass their safety
## Round ID:
N/A

## Reproduction:
image
Step one: Grab duelling pistol from one of the derelicts
Step two: Create a turret, place your duelling pistol within your constructed turret
Step three: Finish construction of your turret
Step four: Get your head blown off by the turret, which bypasses the safety and instakills you (and will continue to instakill anyone else that the turret is set to target)

Duelling pistols will just vaporize your head with no recourse, no brain, no gib - for context!

### PR DESCRIPTION
**Pull request by Thunder12345**
## About The Pull Request

Fixes #75197

Adds a global list of gun types banned from being added to turrets. Adds dueling pistols to the banned list as they can freely round remove people when in turrets without the usual restrictions.

## Why It's Good For The Game

Instakill turrets are an undesirable outcome.

## Changelog
:cl:
fix: Dueling pistols can no longer be fitted in turrets to instantly kill people.
/:cl:

### CODE REVIEW

File: code/__DEFINES/is_helpers.dm [hunk 1]
```diff
@@ -269,6 +269,12 @@ GLOBAL_LIST_INIT(book_types, typecacheof(list(
 	/obj/item/spellbook,
 	/obj/item/storage/book)))
 
+#define isForbiddenTurretGun(O) (is_type_in_typecache(O, GLOB.forbidden_turret_guns))
```
Comment (Fikou) [inline code/__DEFINES/is_helpers.dm:272]:
```suggestion
#define is_forbidden_turret_gun(O) (is_type_in_typecache(O, GLOB.forbidden_turret_guns))
```

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by optimumtact** [comment]
this should probably be an allow list instead of a deny list

File: code/game/machinery/porta_turret/portable_turret_construct.dm [hunk 2]
```diff
@@ -99,17 +99,16 @@
 
 		if(PTURRET_INTERNAL_ARMOUR_ON)
 			if(istype(I, /obj/item/gun/energy)) //the gun installation part
-				if(I.gun_flags & TURRET_INCOMPATIBLE)
+				var/obj/item/gun/energy/E = I
+				if(E.gun_flags & TURRET_INCOMPATIBLE)
```
Comment (Fikou) [inline code/game/machinery/porta_turret/portable_turret_construct.dm]: you must use descriptive variable names, not E.

**Review by Fikou — CHANGES_REQUESTED** [review]

**Comment by Fikou** [comment]
good enough

**Pull request merged**